armory/utils/metrics.py

```diff
@@ -151,6 +157,44 @@ def _snr_spectrogram(x_i, x_adv_i):
     return signal_power / noise_power
 
 
+@contextmanager
+def resource_context(name="Name", profiler=None):
+    profiler_types = ["Basic", "Deterministic"]
+    if profiler is not None and profiler not in profiler_types:
+        raise ValueError(f"Profiler {profiler} is not one of {profiler_types}.")
```

Comment:
There are a lot of "if" statements in the function, which make it harder to parse. I would recommend starting with something like:
```
if profiler is None:
    yield
    return 0
profiler_types = ["Basic", "Deterministic"]
...
```
as that will remove a lot of the ifs.